parsec/core/backend_connection/authenticated.py

```diff
@@ -60,7 +70,11 @@ def copy_exception(exception: BaseExceptionTypeVar) -> BaseExceptionTypeVar:
 
 
 class BackendAuthenticatedCmds:
-    def __init__(self, addr: BackendOrganizationAddr, acquire_transport):
+    def __init__(
+        self,
+        addr: BackendOrganizationAddr,
+        acquire_transport: Callable[..., _AsyncGeneratorContextManager[Transport]],
```

Comment:
Maybe `acquire_transport` is a good use case for `Protocol`:
```python
class AcquireTransport(Protocol):
    def __call__(
        self,
        force_fresh: bool = False,
        ignore_status: bool = False,
        allow_not_available: bool = False,
    ) -> Transport:
        ...
```
This way we can type those arguments too